test/extensions/filters/http/header_to_metadata/config_test.cc

```diff
@@ -0,0 +1,74 @@
+#include <string>
+
+#include "envoy/config/filter/http/header_to_metadata/v2/header_to_metadata.pb.validate.h"
+
+#include "extensions/filters/http/header_to_metadata/config.h"
+
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace HeaderToMetadataFilter {
+
+using HeaderToMetadataProtoConfig = envoy::config::filter::http::header_to_metadata::v2::Config;
+
+TEST(HeaderToMetadataFilterConfigTest, InvalidEmptyHeader) {
+  const std::string yaml = R"EOF(
+request_rules:
+- header: ""
+  )EOF";
+
+  HeaderToMetadataProtoConfig proto_config;
+  EXPECT_THROW(TestUtility::loadFromYamlAndValidate(yaml, proto_config), ProtoValidationException);
+}
+
+TEST(HeaderToMetadataFilterConfigTest, InvalidEmptyKey) {
+  const std::string yaml = R"EOF(
+request_rules:
+  - header: x-version
+    on_header_present:
+      metadata_namespace: envoy.lb
+      key: ""
+      type: STRING
+  )EOF";
+
+  HeaderToMetadataProtoConfig proto_config;
+  EXPECT_THROW(TestUtility::loadFromYamlAndValidate(yaml, proto_config), ProtoValidationException);
+}
+
+TEST(HeaderToMetadataFilterConfigTest, SimpleConfig) {
+  const std::string yaml = R"EOF(
+request_rules:
+  - header: x-version
+    on_header_present:
+      metadata_namespace: envoy.lb
+      key: version
+      type: STRING
+    on_header_missing:
+      metadata_namespace: envoy.lb
+      key: default
+      value: 'true'
+      type: STRING
+  )EOF";
+
+  HeaderToMetadataProtoConfig proto_config;
+  TestUtility::loadFromYamlAndValidate(yaml, proto_config);
+
+  testing::NiceMock<Server::Configuration::MockFactoryContext> context;
+  HeaderToMetadataConfig factory;
+
+  Http::FilterFactoryCb cb = factory.createFilterFactoryFromProto(proto_config, "stats", context);
+  Http::MockFilterChainFactoryCallbacks filter_callbacks;
+  EXPECT_CALL(filter_callbacks, addStreamFilter(_));
+  cb(filter_callbacks);
+}
+
+} // namespace HeaderToMetadataFilter
+} // namespace HttpFilters
+} // namespace Extensions
+} // namespace Envoy
```

Comment:
nit: new line at end of file

Reply:
huh, I'd expect clang-format to take care of that, but looks like there's no option for it...

updated.